source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -321,15 +310,11 @@ absl::Status ProcessorState::handleBodyResponse(const BodyResponse& response) {
       partial_body_processed_ = true;
 
       // If anything else is left on the queue, inject it too
-      Buffer::OwnedImpl leftover_data;
-      while (auto leftover_chunk = dequeueStreamingChunk(false, leftover_data)) {
```

Comment:
As the leftover data buffer is stored in received_data_, we can inject it back to filter chain in one shot. No need this while loop any more.